### PR TITLE
Added Calculation Of Mechanical Energy

### PR DESCRIPTION
**Pull request by lostybtw**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: physics/mechanical_energy.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    '759.0712499999998 joules'
+
+
+    """
+    unit = "joules"
+
+    try:
+        kinetic_energy = 0.5 * mass * velocity * velocity
+    except Exception:
+        return -0.0
```
Comment (CaedenPH) [inline physics/mechanical_energy.py:58]:
```suggestion
        return -0.0

```

File: physics/mechanical_energy.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    gravitational_potential_energy = str(gravitational_potential_energy)
+    gravitational_potential_energy += " " + unit
+    return gravitational_potential_energy
+
+
+def calculate_kinetic_energy(mass: float, velocity: float) -> str:
+    """
+
```
Comment (CaedenPH) [inline physics/mechanical_energy.py:37]:
```suggestion
```

File: physics/mechanical_energy.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     energies are related.
+
+    >>> calculate_kinetic_energy(10,20)
+    '2000.0 joules'
+    >>> calculate_kinetic_energy(10,40)
+    '8000.0 joules'
+    >>> calculate_kinetic_energy(4.25,18.9)
+    '759.0712499999998 joules'
+
+
```
Comment (CaedenPH) [inline physics/mechanical_energy.py:51]:
```suggestion
```

File: physics/mechanical_energy.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    and stored in the body, hence potential and
+    kinetic energies are related.
+
+    >>> calculate_gravitational_potential_energy(15,9.8,10)
+    '1470.0 joules'
+    >>> calculate_gravitational_potential_energy(1e100,1e100,1e100)
+    '1e+300 joules'
+    >>> calculate_gravitational_potential_energy(15.5,9.8,25.5)
+    '3873.4500000000003 joules'
+
```
Comment (CaedenPH) [inline physics/mechanical_energy.py:22]:
```suggestion
```

File: physics/mechanical_energy.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    """
+    unit = "joules"
+
+    try:
+        gravitational_potential_energy = mass * gravity * height
+    except Exception:
+        return -0.0
```
Comment (CaedenPH) [inline physics/mechanical_energy.py:29]:
```suggestion
        return -0.0

```

File: physics/mechanical_energy.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    kinetic_energy = str(kinetic_energy)
+    kinetic_energy += " " + unit
+    return kinetic_energy
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
```
Comment (CaedenPH) [inline physics/mechanical_energy.py:67]:
```suggestion
    doctest.testmod()

```

**Review by CaedenPH — APPROVED** [review]

**Comment by lostybtw** [comment]
Can You Merge?

**Comment by lostybtw** [comment]
@CaedenPH can you help me with this?

`sh
mypy --ignore-missing-imports mechanical_energy.py
mechanical_energy.py:28: error: Incompatible types in assignment (expression has type "str", variable has type "float")
mechanical_energy.py:29: error: Unsupported operand types for + ("float" and "str")
mechanical_energy.py:54: error: Incompatible types in assignment (expression has type "str", variable has type "float")
mechanical_energy.py:55: error: Unsupported operand types for + ("float" and "str")
Found 4 errors in 1 file (checked 1 source file)
`

**Comment by CaedenPH** [comment]
> @CaedenPH can you help me with this?
> 
> 
> 
> ```
> 
> mypy --ignore-missing-imports mechanical_energy.py
> 
> mechanical_energy.py:28: error: Incompatible types in assignment (expression has type "str", variable has type "float")
> 
> mechanical_energy.py:29: error: Unsupported operand types for + ("float" and "str")
> 
> mechanical_energy.py:54: error: Incompatible types in assignment (expression has type "str", variable has type "float")
> 
> mechanical_energy.py:55: error: Unsupported operand types for + ("float" and "str")
> 
> Found 4 errors in 1 file (checked 1 source file)
> 
> ```

- [ ] These errors are due to  certain failed return types. For example, you supposedly return `str`, however in one case (in the exception), you return `-0.0`, a `float`. You need to ensure that all possible returns are the return type that you mention. Unless you typehint the return to both float and str, `float | str`

**Comment by lostybtw** [comment]
Thanks! It works now that I thought about it

**Comment by lostybtw** [comment]
Merge Ready?

**Comment by lostybtw** [comment]
@CaedenPH can you merge?

**Comment by lostybtw** [comment]
sorry for ping its just ...

**Review by tianyizheng02 — COMMENTED** [review]
Already implemented by [`kinetic_energy.py`](https://github.com/TheAlgorithms/Python/blob/master/physics/kinetic_energy.py) and [`potential_energy.py`](https://github.com/TheAlgorithms/Python/blob/master/physics/potential_energy.py)

(Yes, I know your PR is older so it probably should've been merged instead, but those two were the ones that the repo owners ended up merging, sorry)

**Pull request closed without merging**